docs/rfcs/004-multi-file-diff.md

```diff
@@ -0,0 +1,67 @@
+# Multi-file Diffs
+
+## Status
+
+Proposed
+
+## Summary
+
+Users can select multiple files in unstaged changes and staged changes pane and see the diffs of selected files in one view, akin to the [`Files changed` tab in pull requests on github.com](https://github.com/atom/github/pull/1753/files).
+
+## Motivation
+
+So that users can view a full set of changes with more context before staging or committing those changes.
+
+The ability to display multiple diffs in one view will also serve as a building block for the following planned features:
+- [commit pane item](#1655) where it shows all changes in a single commit
+- [new PR review flow](https://github.com/atom/github/blob/master/docs/rfcs/003-pull-request-review.md) that shows all changed files proposed in a PR
+
+
+## Explanation
+
+#### Mock-ups coming soon
+
+
+#### Unstaged Changes pane
+- User can `cmd+click` and select multiple files from the list of unstaged changes, and the pane on the left (see multi-file diff section below) will show diffs of the selected files. That pane will continue to reflect any further selecting/unselecting on the Unstaged Changes pane.
+- Once there is at least one file selected, `Stage All` button should be worded as `Stage Selected`.
+
+#### Staged Changes pane
+- Same behavior as Unstaged Changes pane.
+- Once there is at least one file selected, `Unstage All` button should be worded as `Unstage Selected`.
+
+#### Multi-file diff view
+_(note: The following is a summary of what we would like the UX to achieve, but I don't have a clear visuals of what that looks like yet.)_
+
+- Shows diffs of multiple files as a stack.
+- Each diff should show up as its own block, and the current functionality should remain independent of each block.
+- It should be easy to jump quickly to a specific file you care about, or back to the file list to get to another file. Dotcom does so by creating a `jump to` drop down.
+- As user scrolls through a long list of diffs, there should be a sticky heading which remains visible showing the filename of the diff being viewed.
+Nice-to-have UX that doesn't necessarily need to be implemented
+- Each file diff can be collapsed (this can potentially be )
+
+
+## Drawbacks
+
+- `cmd-click` to select multiple files might not be as universally known as we assume, so that might affect discoverability of this feature.
+- There might be performance concerns having to render many diffs at once.
```

Comment:
Hopefully @smashwilson's work on making diffs use a `TextEditor` will make performance a non issue.  Regardless, we should find a way of measuring perf in general and for diff rendering specifically.

Reply:
It should help, depending on the way we implement!

It'd likely be worth it to track down some Giant-Ass Diffs to test this with to suss out edge cases. I'm thinking a Go project where they just took out vendored dependencies, say.

Reply:
Certain diffs could also be loaded only on demand. Like when the diff is very large. Or when the whole file got deleted:

![image](https://user-images.githubusercontent.com/378023/47559897-b827e200-d951-11e8-9da6-772ec77291e6.png)